py_gtfs_rt_ingestion/lib/postgres_utils.py

```diff
@@ -86,35 +141,54 @@ def get_local_engine(echo: bool = False) -> sa.engine.Engine:
         raise exception
 
 
-def insert_metadata(written_file) -> None:  # type: ignore
+def _rds_writer_process(metadata_queue: Queue) -> None:
     """
-    add a row to metadata table containing the filepath and its status as
-    unprocessed
+    process for writing matadata paths recieved from metadata_queue
+
+    if None recieved from queue, process will exit
     """
-    filepath = written_file.path
-    process_logger = ProcessLogger("metadata_insert", filepath=filepath)
-    process_logger.log_start()
+    engine = get_local_engine()
 
-    metadata_table = sa.Table(
-        "metadataLog",
-        sa.MetaData(),
-        sa.Column("pk_id", sa.Integer, primary_key=True),
-        sa.Column("processed", sa.Boolean, default=sa.false()),
-        sa.Column("path", sa.String(256), nullable=False, unique=True),
-        sa.Column(
-            "created_on",
-            sa.DateTime(timezone=True),
-            server_default=sa.func.now(),
-        ),
+    sa.event.listen(
+        engine,
+        "do_connect",
+        postgres_event_update_db_password,
     )
 
-    try:
-        with get_local_engine().connect() as connection:
-            stmt = sa.insert(metadata_table).values(
-                processed=False, path=filepath
-            )
-            connection.execute(stmt)
-            connection.commit()
-        process_logger.log_complete()
-    except Exception as exception:
-        process_logger.log_failure(exception)
+    session = sessionmaker(bind=engine)
+
+    while True:
+        metadata = metadata_queue.get()
+
+        if metadata is None:
+            break
```

Comment:
why not 
`while not metadata_queue.empty():`
?

Reply:
the queue will probably be empty for a majority of the time the process is running.  with `while not metadata_queue.empty():` it would probably just exit the process immediately

Reply:
got it. i didn't realize we were sending `None` to exit.